packages/naming.cell.match/cell-match.test.js

```diff
@@ -0,0 +1,133 @@
+'use strict';
+
+const safeEval = require('node-eval');
+
+const BemCell = require('@bem/sdk.cell');
+const { origin } = require('@bem/sdk.naming.presets');
+const createMatch = require('.');
+
+const flatOriginMatch = createMatch(Object.assign({}, origin, { fs: Object.assign({}, origin.fs, { scheme: 'flat' }) }));
+const originMatch = createMatch(origin);
+const flatModernMatch = createMatch(Object.assign({}, origin, { fs: Object.assign({}, origin.fs, { scheme: 'flat',
+    pattern: '${entity}${layer?@${layer}}.${tech}' }) }));
+
+const { expect } = require('chai');
+
+describe('naming.cell.match', () => {
+    for (const [dTitle, [match, its]] of Object.entries({
+        'flat / origin': [flatOriginMatch, rawses`
+            reject invalid                 → blocks                        → { isMatch: false }
+            reject invalid block: _block   → common.blocks/_button         → { isMatch: false }
+            reject invalid block: .button  → common.blocks/.button         → { isMatch: false }
+            reject nested scheme           → common.blocks/button/_mod     → { isMatch: false }
+            reject block without tech      → common.blocks/button/button   → { isMatch: false }
+            match valid block: button      → common.blocks/button          → { isMatch: true }
+            match partial layer            → common.blocks                 → { isMatch: true }
+            parse fully qualified tech     → common.blocks/bb/bb.css       → { cell: { layer: 'common', block: 'bb', tech: 'css' } }
+
+            parse full path to block       → dd.blocks/bb/bb.t             → { cell: { layer: 'dd', block: 'bb', tech: 't' } }
+            parse full path to block mod   → dd.blocks/bb/bb_m.t           → { cell: { layer: 'dd', block: 'bb', mod: 'm', tech: 't' } }
+            parse full path to block mod2  → dd.blocks/bb/bb_m_v.t         → { cell: { layer: 'dd', block: 'bb', mod: 'm', val: 'v', tech: 't' } }
+            parse full path to elem        → dd.blocks/bb/bb__e.t          → { cell: { layer: 'dd', block: 'bb', elem: 'e', tech: 't' } }
+            parse full path to elem mod    → dd.blocks/bb/bb__e_m.t        → { cell: { layer: 'dd', block: 'bb', elem: 'e', mod: 'm', tech: 't' } }
+            parse full path to elem mod2   → dd.blocks/bb/bb__e_m_v.t      → { cell: { layer: 'dd', block: 'bb', elem: 'e', mod: 'm', val: 'v', tech: 't' } }
+
+            rejects alien block            → dd.blocks/qq/bb.t               → { isMatch: false }
+            rejects alien block mod        → dd.blocks/qq/bb_m.t          → { isMatch: false }
+            rejects alien block mod2       → dd.blocks/qq/bb_m_v.t        → { isMatch: false }
+            rejects alien elem             → dd.blocks/qq/bb__e.t        → { isMatch: false }
+            rejects alien elem mod         → dd.blocks/qq/bb__e_m.t   → { isMatch: false }
+            rejects alien elem mod2        → dd.blocks/qq/bb__e_m_v.t → { isMatch: false }
```

Comment:
Something went wrong with tabs before the arrow

Reply:
They are spaces ;-D